File: frontend/src/types/product.ts
export interface DeviceType {
  id: number;
  name: string;
  slug: string;
  description: string;
  icon: string | null;
  domain: "COMPUTERS" | "PHONES";
}

export interface Brand {
  id: number;
  name: string;
}

export interface Series {
  id: number;
  name: string;
  brand: number; // ID
  device_type: number; // ID
  description: string;
}

export interface ProductModel {
  id: number;
  name: string;
  brand: number; // ID
  series: number; // ID
}

export interface Product {
  id: number;
  name: string;
  ean13: string | null;
  sku: string | null;
  serial_number: string;
  image_url: string;
  price: string;
  repair_price: string;
  special_price: string;
  other_price: string;
  brand: number | null; // ID
  model: number | null; // ID
  created_at: string;
  updated_at: string;
}

File: backend/apps/tech/models/product.py
from decimal import Decimal

from django.db import models

from .brand import Brand
from .product_model import ProductModel


# Note: Using a string reference for 'repairs.Repair' as the app/model may not be defined yet.


class Product(models.Model):
    """
    Represents a product or part for inventory and sale.
    """

    name = models.TextField(verbose_name="Product Name")
    ean13 = models.CharField(
        max_length=13, unique=True, blank=True, null=True, verbose_name="EAN-13 Barcode"
    )
    sku = models.CharField(
        max_length=100, unique=True, blank=True, null=True, verbose_name="SKU"
    )
    serial_number = models.CharField(max_length=100, blank=True, verbose_name="Serie")
    image_url = models.URLField(blank=True, verbose_name="Image URL")

    # Pricing
    price = models.DecimalField(
        max_digits=10,
        decimal_places=2,
        default=Decimal("0.00"),
        verbose_name="Retail Price",
    )
    repair_price = models.DecimalField(
        max_digits=10,
        decimal_places=2,
        default=Decimal("0.00"),
        verbose_name="Repair Price",
    )
    special_price = models.DecimalField(
        max_digits=10,
        decimal_places=2,
        default=Decimal("0.00"),
        verbose_name="Special Price",
    )
    other_price = models.DecimalField(
        max_digits=10,
        decimal_places=2,
        default=Decimal("0.00"),
        verbose_name="Other Price",
    )

    # Relationships
    brand = models.ForeignKey(
        Brand,
        on_delete=models.SET_NULL,
        null=True,
        blank=True,
        related_name="products",
        verbose_name="Brand",
    )
    model = models.ForeignKey(
        ProductModel,
        on_delete=models.SET_NULL,
        null=True,
        blank=True,
        related_name="products",
        verbose_name="Model",
    )

    # Timestamps
    created_at = models.DateTimeField(auto_now_add=True)
    updated_at = models.DateTimeField(auto_now=True)

    class Meta:
        verbose_name = "Product"
        verbose_name_plural = "Products"
        ordering = ["name"]

    def __str__(self):
        return self.name
